src/adapter/src/coord.rs

```diff
@@ -2323,7 +2328,10 @@ impl Coordinator {
                     !max_compaction_frontier.is_empty(),
                     "`max_compaction_frontier` unexpectedly empty",
                 );
-                max_compaction_frontier
+                // We hold back the `as_of` to near the current time. This is needed for indexes on
+                // REFRESH MVs, because otherwise the `as_of` would be near the next refresh time,
+                // making the index unreadable until then.
+                max_compaction_frontier.meet(&Antichain::from_elem(local_read_ts))
```

Comment:
Can you extend the comment with a justification for why this doesn't break compute reconciliation, like the compaction frontier comment above does?

I _think_ it is: Adapter maintained read holds at the read ts on each index before the restart, and the read ts monotonically increases (even across restarts), so the controller should not have allowed the replicas to compact the indexes beyond the read ts. This might be wrong though. In particular, I'm unsure whether:
* `local_read_ts` is the same timestamp adapter uses for the index read holds.
* `local_read_ts` cannot go backwards between envd restarts.

Reply:
Yes, I've extended the comment as follows:
```
// This is fine for Compute reconciliation, because Adapter was keeping a read hold
// on indexes before the restart at the Oracle read ts (see `advance_timelines`),
// and the Oracle read ts can't go backwards between restarts.
```

> I'm unsure whether:
> local_read_ts is the same timestamp adapter uses for the index read holds.

I think it is, [see in `advance_timelines`](https://github.com/MaterializeInc/materialize/blob/0cc330823e4175a864bd6c0d429472c849b8b69d/src/adapter/src/coord/timeline.rs#L650). (Maybe we'll need to update this once we have multiple timestamp oracles.)

> local_read_ts cannot go backwards between envd restarts.

That has to be true, otherwise we wouldn't be able to provide strict serializability.

Btw. `workflow_test_compute_reconciliation_reuse` has an index on a REFRESH materialized view.